### DISCUSSION TITLE
Grab metadata from a file periodically and update output

### DISCUSSION THREAD

**Discussion by asymcon** [savonet/liquidsoap#2198] [User Support]
Hi,
I'm trying to figure out if there is a way to update song name from a file (output of Now Playing Simple foobar plugin). 
I've been checking the metadata page on the wiki but got stuck: https://www.liquidsoap.info/doc-1.4.4/metadata.html

Basically the idea is to check this file and update 'title' and 'album' based on the formatting of the file:
[album]
[title]

Is such thing possible with liquidsoap?

**Comment by toots** [comment]
Hi!

I believe that, if you have a filename, you can do:
```
r = request.create("/path/to/file")
if request.resolve(r) then
  m = request.metadata(r)
  ..
end
```
This should give you the request metadata. I'd make sure to run this code inside a `thread.run` callback so that it's not blocking the main streaming thread.
